File: top-interview/array/two-sum/two_sum.go
package twosum

func twoSum(nums []int, target int) []int {
	hash := map[int]int{}

	for idx, val := range nums {
		hash[target-val] = idx + 1
	}

	for idx, val := range nums {
		if hash[val] != 0 && idx+1 != hash[val] {
			return []int{idx, hash[val] - 1}
		}
	}
	return []int{}
}

File: top-interview/array/two-sum/two_sum.rs
impl Solution {
  pub fn two_sum(nums: Vec<i32>, target: i32)-> Vec<i32> {
    use std::collections::HashMap;

    let mut hm: HashMap<i32, i32> = HashMap::new();

    for i in 0..nums.len() {
      match hm.get(&(target - nums[i])) {
        Some(&i2) => return vec![i as i32, i2],
        None => hm.insert(nums[i], i as i32),
      };
    }
    return vec![];
  }
}